plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts

```diff
@@ -50,7 +50,7 @@ export async function defaultUserTransformer(
     return undefined;
   }
 
-  const name = normalizeEntityName(user.mail);
+  const name = user.mail.split('@')[0];
```

Comment:
I would suggest to use `normalizeEntityName(user.mail)` as it is the normalization for the mail prefix might still be needed. Or do you see an issue with the auth provider and the matching then?

Reply:
Indeed. It was the main reason for changing it. I see in the [microsoft auth provider code](https://github.com/backstage/backstage/blob/master/plugins/auth-backend/src/providers/microsoft/provider.ts#L236) that, like all other auth providers, the matching id is `userId = profile.email.split('@')[0]`. This would be the easiest way to reconcile the entity ownership of a group member to the current authenticated user.
Of course, I'm open to any suggestion, if there is a better way to get the same result.